### ISSUE TITLE
HTTP server on unix socket errors on close, unlike tcp equivalent

### ISSUE THREAD

**Issue by teleclimber** [denoland/deno#13033]
This is a followup of #8272. The server on unix socket no longer hangs, but it does throw an error. The "tcp" version does not behave like this.

Ideally unix domain socket servers would not throw an error on close.

Here is a complete sample that illustrates the difference: comment/uncomment the different listeners to observe the difference. Note you need to pass a socket file location as an argument.

```ts
import { serveListener } from "https://deno.land/std@0.117.0/http/server.ts";
import { exists} from "https://deno.land/std@0.117.0/fs/mod.ts";
import { sleep } from "https://deno.land/x/sleep/mod.ts";

const sock_path = Deno.args[Deno.args.length -1];
if( !sock_path || !sock_path.endsWith('.sock') ) throw new Error("socket path should be specified and end with sock");
console.log( sock_path );

if( await exists(sock_path) ) await Deno.remove(sock_path);

const controller = new AbortController();

const listener = Deno.listen({ path: sock_path, transport: "unix" });
// const listener = Deno.listen({port:3000});

serveListener( listener, (request) => {
	console.log("got request");
	return new Response("hello world");
}, {signal: controller.signal} );
 
console.log("server started");

await sleep(1);

console.log("closing server");
// listener.close();
controller.abort();

await sleep(1);
```
Thanks.

**Comment by teleclimber** [comment]
I apologize for not providing the basics.

Deno 1.16.4. The error occurs on Linux and Mac. I haven't tested windows.

The error: 

```
error: Uncaught (in promise) Interrupted: operation canceled
        conn = await listener.accept();
               ^
    at async Listener.accept (deno:ext/net/01_net.js:118:19)
    at async Server.#accept (https://deno.land/std@0.117.0/http/server.ts:432:16)
    at async Server.serve (https://deno.land/std@0.117.0/http/server.ts:193:14)
    at async serveListener (https://deno.land/std@0.117.0/http/server.ts:594:10)
```

**Comment by bnoordhuis** [comment]
Thanks, I was able to reproduce. Fix in #13075.
